Update versions for cmatrix doxygen elixir erlang eudev fish

duckunix opened the pull request:
Updating packages to versions:

- cmatrix:  version '1.2'
- elixir:  version '1.5.3'
- doxygen:  version '1.8.14'
- erlang:  version '20.2'
- eudev:  version '3.2.5'
- fish:  version '2.7.1'

[Supporting: #1585 ]

Fixes #

(let GitHub automatically close an issue when this pull request gets merged)

## Before you submit a pull request
This template is not neccessary when you do simple things like updating packages to the latest version. But in doubt, it's always better so provide some information.

## Description
Provide a description, what your changes do and why they are important

Please link issues and other pull requests connected to this one.

## Addtional information
Mention things we might need to know. Like:

Works properly:
- [x] x86_64
- [ ] aarch64 (reasons why it doesn't)

---

## That's it
Thank you for submitting your pull request.
When done, please delete the parts of this template which you don't need or these, which are only for guidance.

uberhacker commented on packages/elixir.rb:
```diff
@@ -8,16 +8,8 @@ class Elixir < Package
   source_sha256 '70972b844c12bc1a3960136d628ab4f21ca87dd5539c544ebabe41d6c9239ba9'
```
Wasn't there an update available for this package?

uberhacker replied:
OK, I know what the issue is here.  This package was already updated in #1665.  You can remove this package from the PR.

uberhacker commented on packages/eudev.rb:
```diff
@@ -3,30 +3,22 @@
   depends_on 'util_linux'
-  depends_on 'm4'
-  depends_on 'autoconf'
   depends_on 'automake'
```
Duplicate `depends_on 'util_linux'` and `depends_on 'automake'` lines?

uberhacker commented on packages/fish.rb:
```diff
@@ -3,21 +3,13 @@
-     x86_64: '6fe888f23b24020270bbb7c30911b2ae4ffc80689c1b66612a6ba6894a50f3a8',
   })
 
   depends_on 'ncurses'
```
Missing `--prefix=#{CREW_PREFIX}` configure option?

duckunix commented:
yes, there was.  Not sure what happened.  Fixing now.

On Tue, Jan 23, 2018 at 9:25 AM, Ed Reel <notifications@github.com> wrote:

> *@uberhacker* commented on this pull request.
> ------------------------------
>
> In packages/elixir.rb
> <https://github.com/skycocker/chromebrew/pull/1688#discussion_r163276918>:
>
> > @@ -8,16 +8,8 @@ class Elixir < Package
>    source_sha256 '70972b844c12bc1a3960136d628ab4f21ca87dd5539c544ebabe41d6c9239ba9'
>
> Wasn't there an update available for this package?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/skycocker/chromebrew/pull/1688#pullrequestreview-90862990>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADs5nj69pKmP1cwaKelifDnfDSLKdt5ks5tNfnkgaJpZM4RpzFx>
> .
>



-- 
Don Harper

duckunix commented:
Cleaned the dep lines up, and added the --prefix to fish.  Updates coming
in a few.

On Tue, Jan 23, 2018 at 9:28 AM, Ed Reel <notifications@github.com> wrote:

> *@uberhacker* commented on this pull request.
> ------------------------------
>
> In packages/fish.rb
> <https://github.com/skycocker/chromebrew/pull/1688#discussion_r163277843>:
>
> >    })
>    binary_sha256 ({
> -    aarch64: '45bd379b711698286d9461150c18fbf1d01446e61aa75aaf5bfcbc0dab1e1248',
> -     armv7l: '45bd379b711698286d9461150c18fbf1d01446e61aa75aaf5bfcbc0dab1e1248',
> -       i686: '173692480715219af0c366e67f182314a895dcbd816819c6a4b45a4c43da07c1',
> -     x86_64: '6fe888f23b24020270bbb7c30911b2ae4ffc80689c1b66612a6ba6894a50f3a8',
>    })
>
>    depends_on 'ncurses'
>
> Missing --prefix=#{CREW_PREFIX} configure option?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/skycocker/chromebrew/pull/1688#pullrequestreview-90864107>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADs5pEfsDvuwyC-rb9A5eVdmp2P2HVpks5tNfqIgaJpZM4RpzFx>
> .
>



-- 
Don Harper

uberhacker commented on packages/doxygen.rb:
```diff
@@ -3,25 +3,16 @@
+  depends_on 'cmake' => :build
 
   def self.build
     system "cmake ."
```
Can we add `-DCMAKE_INSTALL_PREFIX=#{CREW_DEST_PREFIX}` cmake option?

uberhacker commented on packages/cmatrix.rb:
```diff
@@ -3,21 +3,13 @@
-     x86_64: 'a846151b32138a8b0803b3e40dda4d42e804d093dab1105787578714e144131b',
   })
 
   depends_on 'buildessential'
```
You can remove this line since it's already part of crew "core".

duckunix replied:
done

uberhacker commented on packages/eudev.rb:
```diff
@@ -3,30 +3,20 @@
+  depends_on 'util_linux' => :build
 
   def self.build
     system 'autoreconf -f -i -s'
```
Didn't we need the `--libdir=#{CREW_LIB_PREFIX}` configure option since there were library files in this package?

thedamian commented on packages/cmatrix.rb:
```diff
@@ -3,24 +3,15 @@
 class Cmatrix < Package
   description "CMatrix is a program to see the cool scrolling lines from 'The Matrix' movie."
   homepage 'http://www.asty.org/cmatrix/'
-  version '1.2a-1'
```
Why are we moving from 1.2a to 1.2? Isn't that going backwards or is this a cmatrix thing?

thedamian commented:
why is cmatrix going from 1.2a to 1.2 (which I'm guessing that it's in their "archive" version is an "older" version)

uberhacker commented:
@thedamian: If you read the top of [the "official" cmatrix page](http://www.asty.org/cmatrix/), it has this message:
```
Cmatrix has found a new maintainer and repo! Many thanks to Abhishek V Ashok for his interest in maintaining and improving CMatrix. You can find the newest repository for CMatrix at https://github.com/abishekvashok/cmatrix and look for a new official website as well!
```
So I think going forward, we should be using the releases from this repo to get the latest versions.
